### PR TITLE
Implementing DryRun gitprovider

### PR DESCRIPTION
**Pull request by luizbafilho**
Closes: #601 

**What changed?**
- Avoid auth flow and deployment keys upload when dryrun
- Renaming `err` vars to Go convention

**Why?**
- minimize user friction

**How did you test it?**

**Release notes**

**Documentation Changes**

### CODE REVIEW

**Comment by jpellizzari** [comment]
@luizbafilho Does this close #601 ?

**Comment by luizbafilho** [comment]
yes, updated the description

File: pkg/gitproviders/dryrun.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
+package gitproviders
+
+import "github.com/fluxcd/go-git-providers/gitprovider"
+
+type dryrunProvider struct {
+	provider GitProvider
+}
+
+func NewDryRun(_ Config) (GitProvider, error) {
```
Comment (jpellizzari) [inline pkg/gitproviders/dryrun.go:9]: Why do we need a `Config` object as an argument here?

**Review by jpellizzari — COMMENTED** [review]
Is it possible to add test coverage for this?

File: pkg/gitproviders/dryrun.go (hunk 1 above)
Comment (luizbafilho) [reply pkg/gitproviders/dryrun.go:9]: we don't, removing it.

**Comment by luizbafilho** [comment]
> Is it possible to add test coverage for this?

as you mentioned in the issue, is quite hard to require a big refactor, which I think is better done in a separate issue.

But I could add some to the dryrun implementation indeed. Not really testing anything but it will help bump the coverage.

**Comment by jpellizzari** [comment]
> But I could add some to the dryrun implementation indeed. Not really testing anything but it will help bump the coverage.

Not looking to bump the coverage for its own sake, only to prove that this is working programtically. I'll leave it up to you as to whether it makes sense to add some sort of automated test.

**Review by jpellizzari — APPROVED** [review]

**Pull request merged**